### ISSUE TITLE
Fix S4023 FP: Interfaces that provide type parameters from base interface

### ISSUE THREAD

**Issue by bart-vmware** [SonarSource/sonar-dotnet#7633]
### Description

Rule [S4023](https://sonarsource.github.io/rspec/#/rspec/S4023/csharp) is violated if an interface implements another interface, providing generic type parameters, without declaring members of its own.

### Repro steps

```c#
public interface IEndpointHandler<in TArgument, TResult>
{
    Task<TResult> InvokeAsync(TArgument argument, CancellationToken cancellationToken);
}

// FP: S4023
public interface ILoggingEndpointHandler : IEndpointHandler<LoggingRequest, LoggingResponse>
{
}
```

### Expected behavior

No warning, because `ILoggingEndpointHandler` is not a marker interface.

### Actual behavior

Warning S4023.

### Related information

* C#/VB.NET Plugins version: SonarAnalyzer.CSharp v8.47.0.55603
* Visual Studio version: VS 2022 v17.6.5
* MSBuild / dotnet version: .NET 6
* SonarScanner for .NET version (if used)
* Operating System: Windows 11

**Comment by sebastien-marichal** [comment]
Hello @bart-vmware,

Thank you for bringing all the details.

After internal discussion, I confirm this as a false positive.

- [ ] Add bound/constrained generic interfaces as an exception
- [ ] Update the RSPEC:
  - [ ] Add more explanation about what a marker interface is
  - [ ] Add a non-compliant example with reflection code
  - [ ] Add the exception section with both aggregation, bound, and contained interfaces.
